src/gennylib/include/gennylib/ExecutionStrategy.hpp

```diff
@@ -71,19 +72,18 @@ class ExecutionStrategy {
 
         bool shouldContinue = true;
         while (shouldContinue) {
+            auto ctx = _op.start();
             try {
-                auto timer = _timer.start();
-                ++_ops;
                 ++result.numAttempts;
 
-                fun();
-
-                timer.report();
+                fun(ctx);
 
+                ctx.success();
                 result.wasSuccessful = true;
                 shouldContinue = false;
             } catch (const boost::exception& e) {
-                _recordError(e);
+                BOOST_LOG_TRIVIAL(debug) << "Caught error: " << boost::diagnostic_information(e);
+                ctx.fail();
```

Comment:
we start the `ctx` at every retry/iteration so no need to scope-guard close it I think

Reply:
So I agree, but just for posterity, people tend to like scope guards because they're relatively resilient to refactors or unexpected cases.